moto/glue/utils.py

```diff
@@ -0,0 +1,378 @@
+import abc
+import operator
+import re
+import warnings
+from datetime import date, datetime, timedelta
+from typing import Any, Dict, List, Optional, Union
+
+from pyparsing import (
+    CaselessKeyword,
+    Forward,
+    OpAssoc,
+    ParserElement,
+    ParseResults,
+    QuotedString,
+    Suppress,
+    Word,
+    alphanums,
+    delimited_list,
+    exceptions,
+    infix_notation,
+    one_of,
+    pyparsing_common,
+)
+
+from .exceptions import InvalidInputException, InvalidStateException
+
+
+def _cast(type_: str, value: Any) -> Union[date, datetime, float, int, str]:
+    # values are always cast from string to target type
+    value = str(value)
+
+    if type_ in ("bigint", "int", "smallint", "tinyint"):
+        try:
+            return int(value)  # no size is enforced
+        except ValueError:
+            raise ValueError(f'"{value}" is not an integer.')
+
+    if type_ == "decimal":
+        try:
+            return float(value)
+        except ValueError:
+            raise ValueError(f"{value} is not a decimal.")
+
+    if type_ in ("char", "string", "varchar"):
+        return value  # no length is enforced
+
+    if type_ == "date":
+        try:
+            return datetime.strptime(value, "%Y-%m-%d").date()
+        except ValueError:
+            # NOTE AWS accepts up until 31st of every month, e.g. '2022-02-31'
+            warnings.warn("Date filtering beyond last of month not supported")
```

Comment:
Is this a relevant warning here? Looks to me like this warning will only be thrown if the date format is unknown, but we never get to the point where the date is valid but unsupported

Reply:
Yes, you're right. I wanted to make the user aware of the fact that their date may be wrong but actually supported by AWS, e.g. February 31st. In that case their production code would work, but the tests using moto would not. Hence I wanted the warning out there to point them to this potential source of the error.

But I agree, we can remove the warning here, as the ValueError will show the faulty date.